system-contracts/bootloader/bootloader.yul

```diff
@@ -2925,6 +2961,133 @@ object "Bootloader" {
+
+                    let success := call(
+                        gas(),
+                        L2_MESSAGE_ROOT_STORAGE(),
+                        0,
+                        msgRootOffset,
+                        add(132, mul(sidesLength, 32)),
+                        0,
+                        0
+                    )
+
+                    if iszero(success) {
+                        debugLog("Failed to set messageRoot: ", 1)
+                        // kl todo this is still failing, as we set the same messageRoot multiple times
+                        revertWithReason(FAILED_TO_SET_MESSAGE_ROOT(), 1)
+                    }
+                    debugLog("MsgRoot set successfully", 2)
+                }
+            }
+
+            /// @notice Sends the rolling hash of the dependency message roots to the L1.
+            function sendMsgRootsRollingHashToL1() {
+                debugLog("Sending message roots to L1", 0)
+                let msgRootSlot := MESSAGE_ROOT_BEGIN_SLOT()
+                let msgRootSlotSize := MESSAGE_ROOT_SLOT_SIZE() 
+                let rollingHashOfProcessedRoots := 0
+                for {let i := 0} true {i := add(i, 1)} {
+                    let messageRootStartSlot := add(32, mul(add(msgRootSlot, mul(i, msgRootSlotSize)), 32))
+                    let chainId  := mload(messageRootStartSlot) 
+                    let blockNumber := mload(add(messageRootStartSlot, 32))
+                    let sidesLength := mload(add(messageRootStartSlot, 64))
+
+                    debugLog("Send roots L1 chainId     ", chainId)
+                    debugLog("Send roots L1 blockNumber ", blockNumber)
+                    debugLog("Send roots L1 sidesLength ", sidesLength)
+
+                    if iszero(sidesLength) {
+                        // There are no more logs, sending hash to L1.
+                        debugLog("MsgRoot hash to L1", rollingHashOfProcessedRoots)
+                        sendToL1Native(true, messageRootRollingHashLogKey(), rollingHashOfProcessedRoots)
+                        break
+                    }
+
+                    let msgRootOffset := 64
+                    mstore(msgRootOffset, {{RIGHT_PADDED_SET_L2_MESSAGE_ROOT_SELECTOR}}) // todo
+                    mstore(add(msgRootOffset, 4), chainId)
+                    mstore(add(msgRootOffset, 36), blockNumber)
+                    mstore(add(msgRootOffset, 68), 96)
+                    mstore(add(msgRootOffset, 100), sidesLength)
+                    for {let j := 0} lt(j, sidesLength) {j := add(j, 1)} {
+                        mstore(add(add(msgRootOffset, 132), mul(j, 32)), mload(add(messageRootStartSlot, mul(add(3, j), 32))))
+                    }
+
+                    // for single messageRoots that are not really sides, we send them to L1 here.
+                    if lt(sidesLength, 2) {
+                        // Calculate keccak256 of all data
+                        mstore(36, rollingHashOfProcessedRoots)
```

Comment:
When you write to memory locatoin `36`, locations `[36..68)` will be filled. This will overwrite the selector that was written above. Probably it works because the selector is not needed anyway, but in this case it is better to just remove it altogether to avoid confusion